setup-dev.sh

```diff
@@ -5,6 +5,25 @@ echo "# Install testing tools."
 echo "# Only works with Ubuntu / APT."
 echo "# --------------------------------------"
 
+hook_script_name=".git/hooks/pre-push"
+
+if [ -t 1 ]; then
+    while true; do
+        echo "Install a git hook to automatically lint files before pushing? ([y]/N)"
+        read yn
+        case $yn in
+            [Yy]* | "")
+                echo "./lint.sh -f" >>$hook_script_name
+                echo "pre-push hook installed to $hook_script_name"
+                break
+                ;;
+            [Nn]*)
+                break
+                ;;
+        esac
```

Comment:
Since we want `N` to be the default, and by default we don't need to do anything, we can simplify this to (untested):
```bash
if [[ "$yn" == [Yy]* ]]; then
...
fi
```